### PR TITLE
http_regen_vtt lambda added

### PR DESCRIPTION
**Pull request by aaronshiel**

### CODE REVIEW

File: api.py [hunk 1]
```diff
@@ -249,3 +253,96 @@ def upload_task_status_update(req: UpdateTaskStatusRequest) -> None:
     tdjson = res.json()
     if "errors" in tdjson:
         raise Exception(json.dumps(tdjson.get("errors")))
+
+
+def validate_json(json_data, json_schema):
+    try:
+        jsonschema.validate(instance=json_data, schema=json_schema)
+    except jsonschema.exceptions.ValidationError as err:
+        log.error(msg=err)
```
Comment (mbsimonovic) [inline api.py:262]: is this the correct way to log errors? ive been doing this `log.error(err)`

File: serverless.yml [hunk 2]
```diff
@@ -236,6 +236,24 @@ functions:
...
+          cors: true
+          authorizer:
+            name: authorizer_func
+            resultTtlInSeconds: 300
+            identitySource: method.request.header.Authorization
+            type: token
+    layers:
+      # binaries gets named "Binaries"+"LambdaLayer":
+      # see https://www.serverless.com/framework/docs/providers/aws/guide/layers#using-your-layers
+      - { Ref: BinariesLambdaLayer }
```
Comment (mbsimonovic) [inline serverless.yml:256]: do you need these binaries (ffmpeg, ...)?

File: __events__/regen-vtt-event.json.dist [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      "userArn": null,
+      "userAgent": "curl/7.79.1",
+      "user": null
+    },
+    "domainName": "6111111111.execute-api.us-east-1.amazonaws.com",
+    "apiId": "6111111111"
+  },
+  "body": "{\"mentor\":\"60bad2ab733e6a54b909a351\",\"question\":\"607766dec525ba87bf68e79b\"}",
+  "isBase64Encoded": false
+}
```
Comment (mbsimonovic) [inline __events__/regen-vtt-event.json.dist]: newline missing

File: regen-vtt.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        (
+            transcript,
+            video_media,
+        ) = fetch_answer_transcript_and_media(mentor, question)
+        transcript_to_vtt(video_media["url"], vtt_file_path, transcript)
+        video_path_base = f"videos/{mentor}/{question}/"
+        if os.path.isfile(vtt_file_path):
+            item_path = f"{video_path_base}en.vtt"
+            s3.upload_file(
+                str(vtt_file_path),
```
Comment (mbsimonovic) [inline regen-vtt.py:54]:
```
f"videos/{mentor}/{question}/en.vtt"
```
seems easier than having 2 variables (path_base and item_path)

**Review by mbsimonovic — APPROVED** [review]

File: api.py (hunk 1 above)
Comment (aaronshiel) [reply api.py:262]: effectively the same thing

**Pull request merged**